Models/MovieDetails.swift

```diff
@@ -26,10 +26,32 @@ struct MovieDetails: Codable {
         case runtime
         case overview
     }
+    
+    init(from decoder: Decoder) throws {
+        let values = try decoder.container(keyedBy: CodingKeys.self)
+        
+        title = try values.decode(String.self, forKey: .title)
+        
+        let posterPathString = try values.decode(String.self, forKey: .posterPath)
+        posterPath = URL(string: "https://image.tmdb.org/t/p/w500" + posterPathString)
+        
+        spokenLanguages = try values.decode([Language].self, forKey: .spokenLanguages)
+        genres = try values.decode([Genre].self, forKey: .genres)
+        
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateStyle = .medium
+        dateFormatter.timeStyle = .none
+        dateFormatter.locale = Locale(identifier: "en_US")
+        releaseDate = dateFormatter.string(from: try values.decode(Date.self, forKey: .releaseDate))
+        
+        let runtimeInt = try values.decode(Int.self, forKey: .runtime)
```

Comment:
Same as poster path, this property can be null from [here](https://developers.themoviedb.org/3/movies/get-movie-details), so `decodeIfPresent`